### PR TITLE
feat(wallet) Add stripe onramp

### PR DESCRIPTION
**Pull request by muliswilliam**
Resolves https://github.com/brave/brave-browser/issues/31397

## Submitter Checklist:

- [x] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

Uploading Screen Recording 2023-06-30 at 12.36.05.mov…

### CODE REVIEW

File: components/brave_wallet_ui/assets/svg-icons/stripe-logo.svg [hunk 1]
```diff
@@ -0,0 +1,6 @@
+<svg width="40" height="40" viewBox="0 0 40 40" fill="none" xmlns="http://www.w3.org/2000/svg">
+<g id="Frame 3294">
```
[the remark this answers is not in the document]
Comment (thypon) [reply components/brave_wallet_ui/assets/svg-icons/stripe-logo.svg:2]: I think the issue here is that ID cannot contain spaces

[the remark this answers is not in the document]
Comment (muliswilliam) [reply components/brave_wallet_ui/assets/svg-icons/stripe-logo.svg]: @thypon Removed space https://github.com/brave/brave-core/pull/19112/commits/41dbd02b7492edea59af0eb5d8923599428439b8

**Review by mkarolin — APPROVED** [review]
`strings`++

**Review by Douglashdaniel — APPROVED** [review]
++ Front-end

**Review by nvonpentz — APPROVED** [review]

**Comment by srirambv** [comment]
Verification passed on 

Brave | 1.58.46 Chromium: 115.0.5790.136 (Official Build) nightly (64-bit)
-- | --
Revision | bcdad692fb6e67f67e0214173b0c6f5780443940
OS | Linux

- Verified Stripe is only shown as a purchase option for US
- Verified Stripe is not shown as a purchase option for non-US users

US | Non-US
-- | --
<video src=https://github.com/brave/brave-core/assets/17010094/e5c9fb45-dc93-45c0-a810-a21146e67d9e> | <video src=https://github.com/brave/brave-core/assets/17010094/17d7adf3-65ad-4439-82ad-2195365dba08>

**Pull request merged**